### PR TITLE
AP_Proximity: Fix Cygbot D1 driver CRC

### PR DESCRIPTION
**Pull request by rishabsingh3003**
Fixes the CRC that I couldn't get working in https://github.com/ArduPilot/ardupilot/pull/19162
Turns out to be a silly mistake

### CODE REVIEW

**Comment by rishabsingh3003** [comment]
Fixes: #19427

File: libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp [hunk 1]
```diff
@@ -97,26 +97,24 @@ bool AP_Proximity_Cygbot_D1::parse_byte(uint8_t data)
             _parse_state = Payload_Data;
             _msg.payload_counter = 0;
             _msg.payload[_msg.payload_counter] = data;
-            _msg.payload_counter ++;
```
Comment (rmackay9) [inline libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp:100]: It's a bit odd to have the payload_counter be zero but still use payload[0].  Doesn't this mean that there's no way for the payload to be empty?  I suspect it works it's just a little different from how we normally handle buffers.  Normally the counter or len parameter will be "1" when there is just a single element in the buffer at index 0.

File: libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp [hunk 2]
```diff
@@ -97,26 +97,24 @@ bool AP_Proximity_Cygbot_D1::parse_byte(uint8_t data)
...
 
     case CheckSum: {
-        // To-DO: FIX CHECKSUM below. It does not pass correctly
-        // const uint8_t checksum_num = calc_checksum(_msg.payload, _msg.payload_len);
-        // if (data != checksum_num) {
-        // 	return false;
-        // }
+        const uint8_t checksum_num = calc_checksum(_msg.payload, _msg.payload_len);
+        if (data != checksum_num) {
+        	return false;
```
Comment (rmackay9) [inline libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp:116]: nitpick: minor whitespace issue..

**Comment by rmackay9** [comment]
BTW, you could consider using valgrind to check for sure that it doesn't write past the end of the array (in SITL).

**Pull request merged**